### PR TITLE
Upgrade to Scala.js 1.0.0-RC1

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -31,18 +27,19 @@ val commonSettings: Seq[Setting[_]] = Seq(
...
+)
 
-val previousArtifactSetting: Setting[_] = {
-  previousArtifact := {
+val previousArtifactSetting = Def.settings()
+  /* MiMa is completely disabled while we are in 1.0.0-SNAPSHOT.
+  mimaPreviousArtifacts ++= {
     val scalaV = scalaVersion.value
     val scalaBinaryV = scalaBinaryVersion.value
     if (!scalaVersionsUsedForPublishing.contains(scalaV)) {
```
Comment (sjrd) [inline build.sbt]: Since you removed `scalaVersionsUsedForPublishing`, perhaps also remove this test?

**Review by sjrd — CHANGES_REQUESTED** [review]

File: seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumRun.scala [hunk 2]
```diff
@@ -47,17 +47,22 @@ private sealed class SeleniumRun(
...
+    data.get("consoleLog").forEach(consumer(streams.out.println _))
+    data.get("consoleError").forEach(consumer(streams.err.println _))
+    data.get("msgs").forEach(consumer(receivedMessage _))
 
-    val errs = data("errors").asScala
-    if (errs.nonEmpty) throw new SeleniumRun.WindowOnErrorException(errs.toList)
+    val errs = data.get("errors")
+    if (!errs.isEmpty()) {
+      // Convoluted way of writing errs.toList without JavaConverters.
+      val errList = List(errs.toArray(Array[String]()) :_*)
```
Comment (sjrd) [inline seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumRun.scala]:
```suggestion
      val errList = errs.toArray(Array[String]()).toList
```
?

**Review by sjrd — APPROVED** [review]

**Comment by gzm0** [comment]
Updated (fix in `KeepAliveTest`)

File: seleniumJSEnv/src/test/scala/org/scalajs/jsenv/selenium/KeepAliveTest.scala [hunk 3]
```diff
@@ -24,7 +22,7 @@ class KeepAliveTest {
     def quit(): Unit = closed = true
 
     def executeScript(code: String, args: Object*): Object =
-      Map.empty[String, String].asJava
+      new java.util.HashMap[String, java.util.List[String]]()
```
Comment (sjrd) [inline seleniumJSEnv/src/test/scala/org/scalajs/jsenv/selenium/KeepAliveTest.scala:25]: Hum, why did the type of values change from `String` to `java.util.List[String]`?

Comment (gzm0) [reply seleniumJSEnv/src/test/scala/org/scalajs/jsenv/selenium/KeepAliveTest.scala:25]: Was a bug :) But it's of course not distinguishable for an empty map...

**Comment by gzm0** [comment]
Fixed another omission in the scala version upgrade commit: There were unnecessary if blocks.

**Comment by gzm0** [comment]
Also see: https://github.com/scala-js/scala-js/issues/3884

File: seleniumJSEnvTest/src/test/scala/org/scalajs/jsenv/selenium/StructureTest.scala [hunk 4]
```diff
@@ -22,14 +21,15 @@ class StructureTest {
...
 
   def assertStructure(): Unit = {
-    val nodes = global.document.body.childNodes.asInstanceOf[js.Array[js.Dynamic]]
-    val elements = nodes.takeRight(2) // Ignore scripts
+    val nodes = global.document.body.childNodes
 
-    val h1 = elements(0)
+    // Nodes start at 2 to ignore scripts.
+
+    val h1 = nodes.item(2)
```
Comment (sjrd) [inline seleniumJSEnvTest/src/test/scala/org/scalajs/jsenv/selenium/StructureTest.scala]:
```suggestion
    val h1 = nodes.item(nodes.length - 2)
```

File: seleniumJSEnvTest/src/test/scala/org/scalajs/jsenv/selenium/StructureTest.scala [hunk 5]
```diff
@@ -22,14 +21,15 @@ class StructureTest {
...
     assertEquals("H1", h1.tagName.asInstanceOf[String])
     assertEquals("SeleniumJSEnv Test", h1.textContent.asInstanceOf[String])
 
-    val canvas = elements(1)
+    val canvas = nodes.item(3)
```
Comment (sjrd) [inline seleniumJSEnvTest/src/test/scala/org/scalajs/jsenv/selenium/StructureTest.scala]:
```suggestion
    val canvas = nodes.item(nodes.length - 1)
```

**Review by sjrd — CHANGES_REQUESTED** [review]
The suggested changes below should bring the new code closer to the semantics of the old code, which I suppose should fix the broken tests.

Also I would remove the following sentence from the commit message:

> This surfaces in 2.13 where takeRight seems to not be optimized.

or at least change it so that it doesn't assume what is not confirmed:

> This surfaces in 2.13 where takeRight seems to be compiled differently.

**Review by sjrd — APPROVED** [review]

**Pull request merged**